pkg/detector/preemption.go

```diff
@@ -113,6 +120,28 @@ func (d *ResourceDetector) preemptPropagationPolicy(resourceTemplate *unstructur
 	return nil
 }
 
+// preemptClusterPropagationPolicyDirectly directly preempts resource template claimed by ClusterPropagationPolicy regardless of priority.
+func (d *ResourceDetector) preemptClusterPropagationPolicyDirectly(resourceTemplate *unstructured.Unstructured, policy *policyv1alpha1.PropagationPolicy) error {
+	claimedPolicyName := util.GetLabelValue(resourceTemplate.GetLabels(), policyv1alpha1.ClusterPropagationPolicyLabel)
+	if claimedPolicyName == "" {
+		return nil
+	}
+
+	clusterWideKey, err := keys.ClusterWideKeyFunc(resourceTemplate)
+	if err != nil {
+		// should not happen.
+		return err
+	}
+	if err := d.ApplyPolicy(resourceTemplate, clusterWideKey, policy, policyv1alpha1.ClusterPropagationPolicyLabel); err != nil {
+		klog.Errorf("Failed to apply new propagation policy(%s/%s) on resource template(%s, kind=%s, %s) directly: %v.", policy.Namespace, policy.Name,
+			resourceTemplate.GetAPIVersion(), resourceTemplate.GetKind(), names.NamespacedKey(resourceTemplate.GetNamespace(), resourceTemplate.GetName()), err)
+		return err
+	}
+	klog.V(4).Infof("Propagation policy(%s/%s) has preempted another cluster propagation policy(%s).",
+		policy.Namespace, policy.Name, claimedPolicyName)
```

Comment:
I believe that it is necessary to add some events, but just printing logs may not be user-friendly for end users.

Reply:
see task details: https://github.com/karmada-io/karmada/issues/3787.
complete in the next pr.